### ISSUE TITLE
Infinite request loop when prefetching data from URL that returns error

### ISSUE THREAD

**Issue by KostkaBrukowa** [reduxjs/redux-toolkit#3942]
Hey! 
I've encountered this problem lately and I was wondering what could I do with that.

I have this piece of code

<details>

```tsx
import * as React from 'react'
import { useGetPokemonByNameQuery } from './services/pokemon'

export default function App() {
  const { isLoading } = useGetPokemonByNameQuery('bulbasaur')

  return (
    <div className="App">
       {isLoading ? 'Loading...' : <Pokemon name={'bulbasaur'} />} 
    </div>
  )
}

export const Pokemon = ({ name }: { name: string }) => {
  const {} = useGetPokemonByNameQuery(name)

  return (
    <div>
      Pokemon
    </div>
  )
}
```

</details>

Link to codesanbox: [LINK](https://codesandbox.io/p/sandbox/redux-toolkit-rtk-query-forked-32p8cv?file=%2Fsrc%2Fservices%2Fpokemon.ts%3A14%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpmpjwor00063b6kz9jr7xjp%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpmpjwor00023b6kqaptvxtd%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpmpjwor00033b6kc6jl22gy%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpmpjwpz000i3b6k9ijvq43n%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B70.59573800086339%252C29.404261999136608%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpmpjwpz000i3b6k9ijvq43n%2522%253A%257B%2522id%2522%253A%2522clpmpjwpz000i3b6k9ijvq43n%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clpmpjwor00023b6kqaptvxtd%2522%253A%257B%2522id%2522%253A%2522clpmpjwor00023b6kqaptvxtd%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clpmpjwor00033b6kc6jl22gy%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clpmpjwor00033b6kc6jl22gy%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Basically the theme here is: 
1. I want to preload some data, therefore I use `useGetPokemonByNameQuery` in parent component
2. When the data is loading I'm not rendering child component.
3. When the data is preloaded I want to display the data in child component using the same hook

And everything works well when the request was succesful. 
Unfortunately when the request fails continuously (like in the sandbox there is wrong URL) there is infinite loop of requests coming from RTK query

I was wondering if you could help me with solution here?

**Comment by usedlife** [comment]
same problem.

**Comment by weihe-grabtaxi** [comment]
is there any update, encountering the same problem.

**Comment by NoorBayari** [comment]
+1

**Comment by LordZardeck** [comment]
Just had some help in discord that explained this to me. The problem with this example is that the component is being unmounted/mounted whenever the request is made. Errors won't be re-tried on re-render, but will on mount. The thought process behind this is if you left the page and came back, you'd expect an attempt to be made again, which is what's happening on unmount/remount. If you instead hide/show the component with css but have it always render, you won't get the infinite retry. 

In my use cases, since the data is alway possible to be undefined, on the component that actually needs the data, in your case the `Pokemon` component, just check if you have the data you need, and if not, just return `null` instead of the rest of the component's contents. 

For example:

```jsx
import * as React from 'react'
import { useGetPokemonByNameQuery } from './services/pokemon'

export default function App() {
  const { isLoading } = useGetPokemonByNameQuery('bulbasaur')

  return (
    <div className="App">
       {isLoading ? 'Loading...' : }
       <Pokemon name={'bulbasaur'} />
    </div>
  )
}

export const Pokemon = ({ name }: { name: string }) => {
  const {data} = useGetPokemonByNameQuery(name)

  if (!data) return null

  return (
    <div>
      Pokemon
    </div>
  )
}
```

**Comment by markerikson** [comment]
Worth noting this to add to the docs, yeah.

Some notes from Lenz:

> that said, RTKQ will only make a request like that on component mount, so unless you unmount and mount repeatedly there won't be extra network requests
